Kudu.Services/Diagnostics/ProfileManager.cs

```diff
@@ -0,0 +1,314 @@
+﻿using Kudu.Contracts.Tracing;
+using Kudu.Core.Infrastructure;
+using Kudu.Core.Tracing;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web.Http;
+
+namespace Kudu.Services.Performance
+{
+    internal static class ProfileManager
```

Comment:
I don't think this should be a static class. I think this should be an instance service following the pattern of all our other services - see App_Start\NinjectServices.cs, where we new them all up and add to the DI container. Then, JobsController just takes it as a ctor param like all the other services.